src/components/Camera/hooks/useCreateEvidence.js

```diff
@@ -0,0 +1,110 @@
+  const originalCameraUrisMap = useStore( state => state.originalCameraUrisMap );
+  const currentObservationIndex = useStore( state => state.currentObservationIndex );
+  const observations = useStore( state => state.observations );
+
+  const numOfObsPhotos = currentObservation?.observationPhotos?.length || 0;
+
+  // Save URIs to camera gallery (if a photo was taken using the app,
+  // we want it accessible in the camera's folder, as if the user has taken those photos
+  // via their own camera app).
+  const savePhotosToCameraGallery = useCallback( async uris => {
+    if ( permissionGranted !== "granted" ) { return; }
+    const savedUris = await Promise.all( uris.map( async uri => {
+      // Find original camera URI of each scaled-down photo
+      const cameraUri = originalCameraUrisMap[uri];
+
+      if ( !cameraUri ) {
+        console.error( `Couldn't find original camera URI for: ${uri}` );
+      }
+      logger.info( "savePhotosToCameraGallery, saving cameraUri: ", cameraUri );
+      return CameraRoll.save( cameraUri, { type: "photo", album: "Camera" } );
+    } ) );
+
+    logger.info( "savePhotosToCameraGallery, savedUris: ", savedUris );
+    // Save these camera roll URIs, so later on observation editor can update
+    // the EXIF metadata of these photos, once we retrieve a location.
+    setCameraRollUris( savedUris );
+  }, [originalCameraUrisMap, setCameraRollUris, permissionGranted] );
+
+  const createObsWithCameraPhotos = useCallback( async ( localFilePaths, localTaxon ) => {
+    const newObservation = await Observation.new( );
+    newObservation.observationPhotos = await ObservationPhoto
+      .createObsPhotosWithPosition( localFilePaths, {
+        position: 0,
+        local: true
+      } );
+
+    if ( localTaxon ) {
+      newObservation.taxon = localTaxon;
+      newObservation.owners_identification_from_vision = true;
+    }
+    setObservations( [newObservation] );
+    logger.info(
+      "createObsWithCameraPhotos, calling savePhotosToCameraGallery with paths: ",
+      localFilePaths
+    );
+
+    return savePhotosToCameraGallery( localFilePaths );
+  }, [savePhotosToCameraGallery, setObservations] );
+
+  const createEvidenceForObsEdit = useCallback( async localTaxon => {
+    if ( addEvidence ) {
+      const obsPhotos = await ObservationPhoto
+        .createObsPhotosWithPosition( evidenceToAdd, {
+          position: numOfObsPhotos,
+          local: true
+        } );
+      const updatedCurrentObservation = Observation
+        .appendObsPhotos( obsPhotos, currentObservation );
+      observations[currentObservationIndex] = updatedCurrentObservation;
+      updateObservations( observations );
+      logger.info(
+        "addCameraPhotosToCurrentObservation, calling savePhotosToCameraGallery with paths: ",
+        evidenceToAdd
+      );
+      return savePhotosToCameraGallery( evidenceToAdd );
+    }
+    return createObsWithCameraPhotos( cameraPreviewUris, localTaxon );
```

Comment:
There is a lot going on here, and I'm not sure this hook makes it easier for me to track it. The hook is named `useCreateEvidence`, but it returns a function that takes a taxon as the only arg... which doesn't seem to have anything to do with the media evidence. It seems more like this hook does a couple different things:

1. Saves new photos to the gallery
2. Create a new obs with new photos (which also saves photos to the gallery)
3. Updates zustand state with all these new values where they can be used by ObsEdit

It might make more sense to me if this was called `usePrepareStateForObsEdit` and exported functions that just do what they're called, like `createObservation` and `createPhotos`. Not sure that's the best approach. As the saying goes, people are usually right when they think something's wrong but almost always wrong when they propose a solution.

Reply:
Ok, for now I renamed this. I agree there's a lot going on in here. The main point of moving everything into hooks for me was to start to separate the logic that happens when a photo is taken from the state of the camera (rotated, zoomed, etc.) since it was tricky to see which parts of the code were necessary for updating the PermissionGate. And now we're no longer reasoning about things like the AR model loading while looking at code for the StandardCamera (as example). But agree that this hook could probably be further simplified and detangled.